xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-preferences/xwiki-platform-notifications-preferences-api/src/main/java/org/xwiki/notifications/preferences/email/NotificationEmailUserPreferenceManager.java

```diff
@@ -40,4 +42,31 @@
      * @return the diff type configured for the given user
      */
     NotificationEmailDiffType getDiffType(String userId);
+
+    /**
+     * @param userReference reference of a user
+     * @return the diff type configured for the given user
+     * @since 14.4.2
+     */
+    NotificationEmailDiffType getDiffType(DocumentReference userReference);
+
+    /**
+     * @return the notification interval configured for the current user
+     * @since 14.4.2
+     */
+    NotificationEmailInterval getInterval();
+
+    /**
+     * @param userId id of a user
+     * @return the notification interval configured for the given user
+     * @since 14.4.2
+     */
+    NotificationEmailInterval getInterval(String userId);
+
+    /**
+     * @param userReference reference of a user
+     * @return the notification interval configured for the given user
+     * @since 14.4.2
+     */
+    NotificationEmailInterval getInterval(DocumentReference userReference);
```

Comment:
Some remarks/questions:
1) cannot you use default methods to not break backward compatibility? Including raising a runtime exception if it cannot be implemented with a default (that's unless the feature is critical in which case there's no solution)
2) I don't think it's the best to provide several signatures, we usually try to have only one
3) As Simon said, we use `UserReference` for new APIs. Then we have bridges/APIs to get the current user reference and to convert from string to user reference.
4) We don't break backward compatibility and in the few cases where there's no choice and we still want to do it, we need to ask and have a vote. See https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HBackwardCompatibility

Reply:
Thanks for the review. The latest commit brings the following improvements:
* It uses `default` methods
* It uses `UserReference`

I'll see what I can do with single method signature and backward compatibility. Also working on XWiki.Notifications.Code.NotificationsEmailPreferencesMacro since the one in this PR is wrong.

Reply:
The usage of `UserReference` requires updating also `org.xwiki.notifications.notifiers.internal.email.IntervalUsersManager` to use `UserReference` instead of `DocumentReference`, which has not been done yet.

Reply:
The latest commit restores usage of `DocumentReference` instead of `UserReference` since I noticed this has an impact on `IntervalUserManager` and I'm not sure if we want to update this class as well (might be ok since it's not an interface though) nor how to proceed to make the conversion in that one in case we go for it. I kept only one method signature for `getDiffType(reference)` and `getInterval(reference)`, I updated `NotificationsEmailPreferencesMacro` and made some manual tests.

Reply:
@vmassol Please let me know if you have any advice / recommendation on how to keep improving the PR until passes, but don't hesitate to take over as I'm not sure the next steps are entirely within my reach! Happy to discuss and to try helping though.

Reply:
Just fixed an issue in `NotificationsEmailPreferencesMacro` to make sure the interval select option values are in upper case, since they are in upper case in enum `NotificationEmailInterval`.